tests/src/test_projects/merkle_proof/tests/functions/process_proof.rs

```diff
@@ -0,0 +1,251 @@
+// TODO: More extensive testing should be added when https://github.com/FuelLabs/fuels-rs/issues/353 is revolved.
```

Comment:
`fuel-merkle` has an [extensive test suite](https://github.com/FuelLabs/fuel-merkle/tree/master/tests-data
) which you might be able to use. 


The idea is for this to be a reference for all merkle tree implementations (Rust, TS, Solidity, Sway...) .

You'd need to write some wrapper code to pass the test cases into a deployed instance of your contract.

Reply:
What I am referring to here is the need to test cases where the proof length is greater than two. This is the current array length and limits my possible tests. I've updated the comment to make that more clear.